### PR TITLE
PIC-2168 remove caseIs for get matchesDetail endpoint

### PR DESCRIPTION
**Pull request by rschallapalli**

### CODE REVIEW

File: src/main/java/uk/gov/justice/probation/courtcaseservice/service/OffenderMatchService.java [hunk 1]
```diff
@@ -87,6 +87,20 @@ public OffenderMatchDetailResponse getOffenderMatchDetailsByCaseIdAndDefendantId
         return OffenderMatchDetailResponse.builder().offenderMatchDetails(offenderMatchDetails).build();
     }
 
+    public OffenderMatchDetailResponse getOffenderMatchDetailsByDefendantId(String defendantId) {
```
Comment (jhevans) [inline src/main/java/uk/gov/justice/probation/courtcaseservice/service/OffenderMatchService.java:90]: This is returning a DTO from the service layer, I appreciate the previous method does the same but in general it's probably preferable to return Entities and have the controller do the mapping so these methods are reusable.

Comment (rschallapalli) [reply src/main/java/uk/gov/justice/probation/courtcaseservice/service/OffenderMatchService.java]: Done, thanks

**Review by vb-moj — DISMISSED** [review]

File: src/main/java/uk/gov/justice/probation/courtcaseservice/service/OffenderMatchService.java [hunk 2]
```diff
@@ -87,6 +88,20 @@ public OffenderMatchDetailResponse getOffenderMatchDetailsByCaseIdAndDefendantId
         return OffenderMatchDetailResponse.builder().offenderMatchDetails(offenderMatchDetails).build();
     }
 
+    public List<OffenderMatchDetail> getOffenderMatchDetailsByDefendantId(String defendantId) {
```
Comment (jhevans) [inline src/main/java/uk/gov/justice/probation/courtcaseservice/service/OffenderMatchService.java:91]: Service returning controller DTO

**Review by jhevans — COMMENTED** [review]
Left a few comments for review

File: src/main/java/uk/gov/justice/probation/courtcaseservice/controller/OffenderMatchesController.java [hunk 3]
```diff
@@ -63,6 +64,14 @@ OffenderMatchDetailResponse getOffenderMatchesDetailByCaseAndDefendantId(@PathVa
         return offenderMatchService.getOffenderMatchDetailsByCaseIdAndDefendantId(caseId, defendantId);
     }
 
+    @Operation(description = "Returns all possible matches found for a given defendant ID")
+    @GetMapping(value = "/defendant/{defendantId}/matchesDetail", produces = APPLICATION_JSON_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    public @ResponseBody
+    OffenderMatchDetailResponse getOffenderMatchesDetailByDefendantId(@PathVariable(value = "defendantId") String defendantId) {
```
[the remark this answers is not in the document]
Comment (rschallapalli) [reply src/main/java/uk/gov/justice/probation/courtcaseservice/controller/OffenderMatchesController.java:71]: Yes, true but every other controller method does this so leaving it for consistency

File: src/main/java/uk/gov/justice/probation/courtcaseservice/service/OffenderMatchService.java (hunk 2 above)
Comment (rschallapalli) [reply src/main/java/uk/gov/justice/probation/courtcaseservice/service/OffenderMatchService.java:91]: This service method, apart from fetching the entities, makes rest calls to offender search API for each offender match fetched from the database enriches the returned entities.. so it is not the entities that are returned to the client but the response from the offender search API that is returned, hence the DTOs

File: src/test/java/uk/gov/justice/probation/courtcaseservice/controller/OffenderMatchesControllerIntTest.java [hunk 4]
```diff
@@ -308,6 +312,109 @@ private void validateBody(ValidatableResponse validatableResponse) {
...
+                    .contentType(APPLICATION_JSON_VALUE)
+                    .when()
+                    .get(path)
+                    .then()
+                    .statusCode(404)
+                    .body("userMessage", equalTo("Defendant 90db99d6-04db-11ec-b2d8-0242ac130002 not found"))
+                    .body("developerMessage", equalTo("Defendant 90db99d6-04db-11ec-b2d8-0242ac130002 not found"));
+        }
+
+        private void validate(ValidatableResponse validatableResponse) {
```
[the remark this answers is not in the document]
Comment (rschallapalli) [reply src/test/java/uk/gov/justice/probation/courtcaseservice/controller/OffenderMatchesControllerIntTest.java:370]: sorry, copy pasta of the nested test class that tests the deprecated matchesDetail with caseId endpoint, will rename

[the remark this answers is not in the document]
Comment (rschallapalli) [reply src/test/java/uk/gov/justice/probation/courtcaseservice/controller/OffenderMatchesControllerIntTest.java]: Removed them

**Review by vb-moj — APPROVED** [review]

**Pull request merged**